platforms/documentation/docs/src/docs/release/notes.md

```diff
@@ -36,12 +36,29 @@ Switch your build to use Gradle @version@ by updating the [wrapper](userguide/gr
 
 See the [Gradle 9.x upgrade guide](userguide/upgrading_version_9.html#changes_@baseVersion@) to learn about deprecations, breaking changes, and other considerations when upgrading to Gradle @version@.
 
-For Java, Groovy, Kotlin, and Android compatibility, see the [full compatibility notes](userguide/compatibility.html).   
+For Java, Groovy, Kotlin, and Android compatibility, see the [full compatibility notes](userguide/compatibility.html).
 
 ## New features and usability improvements
 
 <!-- Do not add breaking changes or deprecations here! Add them to the upgrade guide instead. -->
 
+### Test Metadata Logging
+
+Gradle now allows listening for test metadata events during test execution.
+In the exact same manner as [`TestOutputListener`](api/org/gradle/api/tasks/testing/TestOutputListener.html), a [`TestMetadataListener`](api/org/gradle/api/tasks/testing/TestMetadataListener.html) can be registered to receive metadata events emitted by the test framework during via the new [`Test#addTestMetadataListener(TestMetadataListener)`](dsl/org.gradle.api.tasks.testing.Test.html#addTestMetadataListener(TestMetadataListener)) method.
+
+```kotlin
+test {
+    addTestMetadataListener(object : TestMetadataListener {
+        override fun onMetadata(descriptor: TestDescriptor, event: TestMetadataEvent) {
+            logger.lifecycle("From listener: " + descriptor.toString() + " received event: " + event.toString())
+        }
+    })
+}
```

Comment:
I think this can be simplified and we need to access the task through tasks:
```suggestion
tasks.test {
    addTestMetadataListener { descriptor: TestDescriptor, event: TestMetadataEvent ->
        logger.lifecycle("From listener: " + descriptor + " received event: " + event)
    }
}
```